src/Json.Schema.Validation/Validator.cs

```diff
@@ -181,6 +181,44 @@ private void ValidateString(JValue jValue, JsonSchema schema)
                     AddResult(jValue, ErrorNumber.StringDoesNotMatchPattern, value, schema.Pattern);
                 }
             }
+
+            if(!string.IsNullOrEmpty(schema.Format))
+            {
+                ValidateStringFormat(jValue, schema.Format);
+            }
+        }
+
+        private void ValidateStringFormat(JValue jValue, string format)
+        {
+            string value = jValue.Value<string>();
+            if (string.Compare(format, FormatAttributes.DateTime) == 0)
+            {
+                if (!DateTime.TryParse(value, out DateTime _))
+                {
+                    AddResult(jValue, ErrorNumber.StringDoesNotMatchFormat, format, value);
+                }
+            } 
+            else if(string.Compare(format, FormatAttributes.Uri) == 0)
+            {
+                if (!Uri.IsWellFormedUriString(value, UriKind.Absolute))
```

Comment:
>!Uri.IsWellFormedUriString(value, UriKind.Absolute) [](http://example.com/codeflow?start=20&length=51)

will this .net method about equivalent to what is valid in json schema

Reply:
This API actually tells you whether the literal string is well-formed in the sense of requiring further escaping (and some other syntactic validations).

What you actually want to use here is Uri.TryCreate, if it passes, all is well.

Reply:
OK, based on our offline discussion, you actually think (and this seems reasonable) that the expression of a URI in JSON should be normalized. This is contrary to my suggestion, which is to ensure that a specific URI parsing type (the .NET URI class) can initialize an instance based on the data that's provided.

I think you are probably correct. I'm not sure what the behavior of 'is well formed' is for issues other than encoding. Does it check for dot segments, for example? 

In any case, I'm fine accepting this change as you've authored it and continuing the discussion.

Reply:
I have checked it validates the dot segment paths as well. There are issues with this [API](https://github.com/dotnet/runtime/issues/72632) for but using TryCreate succeeds for every possible values that I tried. There is an open github [issue ](https://github.com/dotnet/runtime/issues/78381) for the same.